fix: BROS-87: Introduce unified zoom behavior for trackpads and mice

nick-skriabin opened the pull request:
Just moving the functionality under a feature flag.

Reason: buggy behavior.
<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nick-skriabin committed:
Hide pinch to zoom functionality under a feature flag

nick-skriabin committed:
Fix zoom behavior (unified zoom)

nick-skriabin commented:
/fmt
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15908346263)

yyassi-heartex commented on web/libs/editor/src/tags/object/Image/Image.js:
```diff
@@ -931,25 +932,25 @@ const Model = types
+        );
+        
+        // Apply the limited zoom change
+        const newZoom = clamp(self.currentZoom * limitedZoomChange, MIN_ZOOM, MAX_ZOOM);
+        const zoomScale = newZoom;
```
is `newZoom` being used anywhere? if not, lets clean up the unneeded variable
```suggestion
        const zoomScale = clamp(self.currentZoom * limitedZoomChange, MIN_ZOOM, MAX_ZOOM);
```

nick-skriabin replied:
i don't think so. makes sense to remove, ty!

yyassi-heartex reviewed (approved):
minor suggestion but other wise looks good

nick-skriabin committed:
Apply pre-commit linters

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15908346263

nick-skriabin committed:
Update web/libs/editor/src/tags/object/Image/Image.js

Co-authored-by: Yousif Yassi <104568407+yyassi-heartex@users.noreply.github.com>

hlomzik commented on web/libs/editor/src/tags/object/Image/Image.js:
```diff
@@ -931,25 +932,24 @@ const Model = types
+
+        // Limit the maximum zoom change per event to prevent aggressive zooming
+        // This prevents users from accidentally zooming too far with a single wheel event
+        const limitedZoomChange = Math.max(
```
you need to use `clamp()` here because you don't limit minimum value

hlomzik replied:
basically you just need to run it two times, with ±MAX_CHANGE and with MIN/MAX

hlomzik reviewed (changes requested):
one important change regarding the goal of the PR, and the rest is good

nick-skriabin committed:
Fix zoom behavior and tests

nick-skriabin commented:
/fmt
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/15913085611)

nick-skriabin committed:
Apply pre-commit linters

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/15913085611

nick-skriabin committed:
Add option to control zoom invertion
